zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/tasks/incident/IncidentUpdateRepositoryIT.java

```diff
@@ -319,4 +348,136 @@ private void setupIncidentUpdates(
+              incidentTemplate.getFullQualifiedName(),
+              IncidentTemplate.KEY,
+              List.of("1"),
+              IncidentEntity.class);
+      assertThat(incidents)
+          .hasSize(1)
+          .extracting(IncidentEntity::getState)
+          .containsExactly(IncidentState.RESOLVED);
+    }
+
+    @Test
+    void shouldBulkUpdateListView() throws PersistenceException, IOException {
+      // given
+      final var repository = createRepository();
+      final var bulk = new IncidentBulkUpdate();
+      final var batchRequest = clientAdapter.createBatchRequest();
+      batchRequest.addWithId(
+          listViewTemplate.getFullQualifiedName(),
+          "1",
+          new ProcessInstanceForListViewEntity().setKey(1).setIncident(false));
+      batchRequest.executeWithRefresh();
+
+      // when
+      bulk.listViewRequests()
+          .put(
+              "1",
+              new DocumentUpdate(
+                  "1",
+                  listViewTemplate.getFullQualifiedName(),
+                  Map.of(ListViewTemplate.INCIDENT, true),
+                  "1"));
+      final var result = repository.bulkUpdate(bulk);
+
+      // then
+      assertThat(result).succeedsWithin(REQUEST_TIMEOUT);
+      final var processInstances =
+          search(
+              listViewTemplate.getFullQualifiedName(),
+              ListViewTemplate.PROCESS_INSTANCE_KEY,
+              List.of("1"),
+              ProcessInstanceForListViewEntity.class);
+      assertThat(processInstances)
+          .hasSize(1)
+          .extracting(ProcessInstanceForListViewEntity::isIncident)
+          .containsExactly(true);
+    }
+
+    @Test
+    void shouldBulkUpdateFlowNodeInstances() throws PersistenceException, IOException {
+      // given
+      final var repository = createRepository();
+      final var bulk = new IncidentBulkUpdate();
+      final var batchRequest = clientAdapter.createBatchRequest();
+      batchRequest.addWithId(
+          flowNodeInstanceTemplate.getFullQualifiedName(),
+          "1",
+          new FlowNodeInstanceEntity().setKey(1).setIncident(false));
+      batchRequest.executeWithRefresh();
+
+      // when
+      bulk.flowNodeInstanceRequests()
+          .put(
+              "1",
+              new DocumentUpdate(
+                  "1",
+                  flowNodeInstanceTemplate.getFullQualifiedName(),
+                  Map.of(FlowNodeInstanceTemplate.INCIDENT, true),
+                  "1"));
+      final var result = repository.bulkUpdate(bulk);
+
+      // then
+      assertThat(result).succeedsWithin(REQUEST_TIMEOUT);
+      final var flowNodes =
+          search(
+              flowNodeInstanceTemplate.getFullQualifiedName(),
+              FlowNodeInstanceTemplate.KEY,
+              List.of("1"),
+              FlowNodeInstanceEntity.class);
+      assertThat(flowNodes)
+          .hasSize(1)
+          .extracting(FlowNodeInstanceEntity::isIncident)
+          .containsExactly(true);
```

Comment:
:thought_balloon: I think we should add a test case when there's more than 1 element for each collection in IncidentBulkUpdate

Reply:
I thought about it - technically it would pass if I did something strange like only updating a single element of the map. I thought maybe it's too simple logic tbh, but I can refactor tests to always have 2 entities :shrug:

Reply:
Rewrote tests to include multiple items.